packages/node/test/integration/secure-deposit.spec.ts

```diff
@@ -39,4 +43,125 @@ describe("Node method follows spec - deposit", () => {
       expect(freeBalanceState[key]).toEqual(One);
     }
   });
+
+  it("has the right balance for both parties after deposits of ERC20 tokens and ETH", async () => {
+    const multisigAddress = await createChannel(nodeA, nodeB);
+    const erc20ContractAddress = global["networkContext"]["DolphinCoin"];
+    const erc20Contract = new Contract(
+      erc20ContractAddress,
+      DolphinCoin.abi,
+      new JsonRpcProvider(global["ganacheURL"])
+    );
+
+    let depositReq = makeDepositRequest(
+      multisigAddress,
+      One,
+      erc20ContractAddress
+    );
+
+    expect(
+      await erc20Contract.functions.balanceOf(await nodeA.signerAddress())
+    ).toEqual(Zero);
+
+    let error;
+    try {
+      await nodeA.router.dispatch(depositReq);
+    } catch (e) {
+      error = e;
+    } finally {
+      expect(error).toEqual(
+        INSUFFICIENT_ERC20_FUNDS(await nodeA.signerAddress())
+      );
+    }
+
+    await transferERC20Tokens(await nodeA.signerAddress());
+    await transferERC20Tokens(await nodeB.signerAddress());
+
+    let preDepositBalance = await provider.getBalance(multisigAddress);
+    await nodeA.router.dispatch(depositReq);
+    await nodeB.router.dispatch(depositReq);
+
+    expect(await provider.getBalance(multisigAddress)).toEqual(
+      preDepositBalance
+    );
+    expect(await erc20Contract.functions.balanceOf(multisigAddress)).toEqual(
+      Two
+    );
+
+    await confirmEthAndERC20FreeBalances(
+      nodeA,
+      multisigAddress,
+      erc20ContractAddress
+    );
+    await confirmEthAndERC20FreeBalances(
+      nodeB,
+      multisigAddress,
+      erc20ContractAddress
+    );
+
+    // now deposits ETH
+
+    depositReq = makeDepositRequest(multisigAddress, One);
+
+    preDepositBalance = await provider.getBalance(multisigAddress);
+    await nodeA.router.dispatch(depositReq);
+    await nodeB.router.dispatch(depositReq);
+
+    expect((await provider.getBalance(multisigAddress)).toNumber()).toEqual(
+      preDepositBalance.add(2).toNumber()
+    );
+
+    const freeBalanceState = await getFreeBalanceState(nodeA, multisigAddress);
+    for (const key in freeBalanceState) {
+      expect(freeBalanceState[key]).toEqual(One);
+    }
+  });
 });
+
+/**
+ * @return the ERC20 token balance of the receiver
+ */
+async function transferERC20Tokens(
+  toAddress: string,
+  tokenAddress: string = global["networkContext"]["DolphinCoin"],
```

Comment:
Is the global object here evaluated at runtime?

Reply:
The global object itself yes, but the `networkContext` members are not.